lib/entice/logic/ai/seek.ex

```diff
@@ -0,0 +1,68 @@
+defmodule Entice.Logic.Seek do
+  alias Entice.Entity
+  alias Entice.Logic.{Seek, Player.Position, Npc, Movement}
+  alias Entice.Utils.Geom.Coord
+
+  defstruct target: nil, aggro_distance: 10, escape_distance: 20
+
+  def register(entity),
+  do: Entity.put_behaviour(entity, Seek.Behaviour, [])
+
+  def register(entity, aggro_distance, escape_distance)
+  when is_integer(aggro_distance) and is_integer(escape_distance),
+  do: Entity.put_behaviour(entity, Seek.Behaviour, %{aggro_distance: aggro_distance, escape_distance: escape_distance})
+
+  def unregister(entity),
+  do: Entity.remove_behaviour(entity, Seek.Behaviour)
+
+  #TODO: Add team attr to determine who should be attacked by whom
+  defmodule Behaviour do
+    use Entice.Entity.Behaviour
+
+    def init(entity, %{aggro_distance: aggro_distance, escape_distance: escape_distance}),
+    do: {:ok, entity |> put_attribute(%Seek{aggro_distance: aggro_distance, escape_distance: escape_distance})}
+
+    def init(entity, _args),
+    do: {:ok, entity |> put_attribute(%Seek{})}
+
+    #No introspection for npcs ;)
+    def handle_event({:entity_change, %{entity_id: eid}}, %Entity{id: eid} = entity),
+    do: {:ok, entity}
+
+    def handle_event({:entity_change, %{changed: %{Position => %Position{coord: mover_coord}}, entity_id: moving_entity_id}}, 
+      %Entity{attributes: %{Position => %Position{coord: my_coord},
+                            Movement => _,
+                            Npc      => %Npc{init_coord: init_coord},
+                            Seek     => %Seek{aggro_distance: aggro_distance, escape_distance: escape_distance, target: target}}} = entity) do
+      case target do
```

Comment:
This seems like a lot of resposibility for a single function, can we split this into sub functions?
(You could also match on the nil and moving_entity_id cases in the function header, but it would probably be hard to spot and hard to understand those two cases then :) )